common/src/main/java/com/wynntils/features/user/inventory/InventoryEmeraldCountFeature.java

```diff
@@ -0,0 +1,188 @@
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureCategory;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.mc.event.ContainerRenderEvent;
+import com.wynntils.mc.objects.CommonColors;
+import com.wynntils.mc.render.FontRenderer;
+import com.wynntils.mc.render.HorizontalAlignment;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.render.VerticalAlignment;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.StringUtils;
+import com.wynntils.wynn.objects.EmeraldSymbols;
+import com.wynntils.wynn.utils.ContainerUtils;
+import java.util.Arrays;
+import java.util.Objects;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.client.gui.screens.inventory.InventoryScreen;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@FeatureInfo(category = FeatureCategory.INVENTORY)
+public class InventoryEmeraldCountFeature extends UserFeature {
+    private static final int TEXTURE_SIZE = 28;
+
+    @Config
+    public EmeraldCountType emeraldCountType = EmeraldCountType.Texture;
+
+    @Config
+    public boolean showInventoryEmeraldCount = true;
+
+    @Config
+    public boolean showContainerEmeraldCount = true;
+
+    @SubscribeEvent
+    public void onContainerRender(ContainerRenderEvent event) {
+        int emeralds = ContainerUtils.getEmeraldCountInContainer(McUtils.containerMenu());
+
+        if (!(event.getScreen() instanceof InventoryScreen)) {
+            emeralds -= ContainerUtils.getEmeraldCountInContainer(McUtils.inventoryMenu());
+        }
+
+        if (emeralds == 0) return;
+
+        switch (emeraldCountType) {
+            case Text -> {
+                renderTextCount(event.getPoseStack(), emeralds);
+            }
+            case Texture -> {
+                renderTexturedCount(event.getPoseStack(), emeralds);
+            }
+        }
+    }
+
+    private void renderTextCount(PoseStack poseStack, int emeralds) {
+        Screen screen = McUtils.mc().screen;
+
+        if (!(screen instanceof AbstractContainerScreen<?> containerScreen)) return;
+
+        poseStack.pushPose();
+        poseStack.translate(0, 0, 200);
+
+        String emeraldText;
+        if (GLFW.glfwGetKey(McUtils.mc().getWindow().getWindow(), GLFW.GLFW_KEY_LEFT_SHIFT) == 1) {
+            emeraldText = String.valueOf(emeralds) + EmeraldSymbols.E;
+        } else {
+            String[] emeraldAmounts = Arrays.stream(calculateEmeraldAmounts(emeralds))
+                    .mapToObj(String::valueOf)
+                    .toArray(String[]::new);
+            emeraldText = emeraldAmounts[2] + EmeraldSymbols.LE + " " + emeraldAmounts[1] + EmeraldSymbols.EB + " "
```

Comment:
And the thing that really got me looking into this is that I think that we should leave out parts that are 0. That is, "53 E" looks better than "0 EB 0 LE 43 E".

And my recommendation here is to not try to over-generalize this for just three cases, but just have a simple

if (emeraldAmounts[0]  > 0) {
  stringBuilder.append(emeraldAmounts[0]).append(" LE")
...
etc